McCorrections/src/LeptonWeights.cpp

```diff
@@ -245,22 +272,87 @@ double LeptonWeights::GetTriggerEfficiency(EventInfo& eventInfo, bool isData) co
+                                                                        eventInfo.GetLeg(1)->decayMode());
 
         double tauSF_2 = 1;
         tauSF_2 = isData ?
             tauTriggerWeight_tauTau->getEfficiencyData(static_cast<float>(eventInfo.GetLeg(2).GetMomentum().pt()),
-                                                                    eventInfo.GetLeg(2)->decayMode()) :
+                                                                          eventInfo.GetLeg(2)->decayMode()) :
             tauTriggerWeight_tauTau->getEfficiencyMC(static_cast<float>(eventInfo.GetLeg(2).GetMomentum().pt()),
-                                                                  eventInfo.GetLeg(2)->decayMode());
+                                                                        eventInfo.GetLeg(2)->decayMode());
 
         return  tauSF_1 * tauSF_2;
     }
 
     throw exception ("channel not allowed");
 }
 
+bool LeptonWeights::ApplyUncertaintyScale(int decay_mode, double pt, double eta, GenLeptonMatch gen_lepton_match,
+                                          LegType leg_type, UncertaintySource unc_source)
+{
+    static const std::map<GenLeptonMatch, std::map<UncertaintySource, std::set<int>>> unc_source_map_dm = {
+         {GenLeptonMatch::Tau, { {  UncertaintySource::TauVSjetSF_DM0, {0} } ,
+                                 {   UncertaintySource::TauVSjetSF_DM1, {1} } ,
+                                 {  UncertaintySource::TauVSjetSF_3prong, {10, 1} } } }
+    };
+
+    static const std::map<GenLeptonMatch, std::map<UncertaintySource, std::pair<double, double>>> unc_source_map_pt = {
+         {GenLeptonMatch::Tau, { {  UncertaintySource::TauVSjetSF_pt20to25, {20, 25} } ,
+                                 {   UncertaintySource::TauVSjetSF_pt25to30,{25, 30} } ,
+                                 {  UncertaintySource::TauVSjetSF_pt30to35, {30, 35} },
+                                 {  UncertaintySource::TauVSjetSF_pt35to40, {35, 40} },
+                                 {  UncertaintySource::TauVSjetSF_ptgt40,   {40, 1000} } } }
+    };
+
+    static const std::map<GenLeptonMatch, std::map<UncertaintySource, std::pair<double, double>>> unc_source_map_eta = {
+         {GenLeptonMatch::Electron, { {  UncertaintySource::TauVSeSF_barrel,        {0, 1.460} } ,
+                                      {  UncertaintySource::TauVSeSF_endcap,        {1.558, 3} } } },
+         {GenLeptonMatch::Muon,     { {  UncertaintySource::TauVSmuSF_etaLt0p4,     {0, 0.4} } ,
+                                      {  UncertaintySource::TauVSmuSF_eta0p4to0p8,  {0.4, 0.8} },
+                                      {  UncertaintySource::TauVSmuSF_eta0p8to1p2,  {0.8, 1.2} },
+                                      {  UncertaintySource::TauVSmuSF_eta1p2to1p7,  {1.2, 1.7} },
+                                      {  UncertaintySource::TauVSmuSF_etaGt1p7,     {1.7, 3} } } },
+    };
+
+    if(gen_lepton_match == GenLeptonMatch::Tau && leg_type == LegType::tau){
```

Comment:
- L319-352 should be inside `if(leg_type == LegType::tau)`
- should be else if for leg types e and mu.